addons/product/models/product_packaging.py

```diff
@@ -33,7 +33,7 @@ def _check_barcode_uniqueness(self):
         """ With GS1 nomenclature, products and packagings use the same pattern. Therefore, we need
         to ensure the uniqueness between products' barcodes and packagings' ones"""
         domain = [('barcode', 'in', [b for b in self.mapped('barcode') if b])]
-        if self.env['product.product'].search(domain, order="id", limit=1):
```

Comment:
`order="id"` was added to solve some performance issue linked to the ordering.
`search_count` doesn't order at all.